engine/gui/src/dmsdk/gui/gui.h

```diff
@@ -170,41 +169,173 @@ namespace dmGui
      */
     HNode NewNode(HScene scene, const dmVMath::Point3& position, const dmVMath::Vector3& size, NodeType node_type, uint32_t custom_type);
 
+    /*#
+     * Defer delete a node
+     * @name DeleteNode
+     * @param scene [type:dmGui::HScene] the gui scene
+     * @param node [type:dmGui::HNode] the node to delete
+     */
+    void DeleteNode(HScene scene, HNode node);
+
+    /*#
+     * Set the id of a node.
+     * @note The id must be unique
+     * @name SetNodeId
+     * @param scene [type:dmGui::HScene] the gui scene
+     * @param node [type:dmGui::HNode] the gui node
+     * @param id [type:dmhash_t] the id
+     */
+    void SetNodeId(HScene scene, HNode node, dmhash_t id);
+
+    /*#
+     * Get the id of a node.
+     * @name GetNodeId
+     * @param scene [type:dmGui::HScene] the gui scene
+     * @param node [type:dmGui::HNode] the gui node
+     * @return id [type:dmhash_t] the id of the node
+     */
+    dmhash_t GetNodeId(HScene scene, HNode node);
+
+    /*#
+     * Set the parent of a gui node
+     * @name SetNodeParent
+     * @param scene [type:dmGui::HScene] the gui scene
+     * @param node [type:dmGui::HNode] the gui node
+     * @param parent [type:dmGui::HNode] the new parent. May be null
+     * @param keep_scene_transform [type:bool] true to keep the world position
+     * @return result [type:dmGui::Result] dmGui::RESULT_OK is successful
+     */
+    Result SetNodeParent(HScene scene, HNode node, HNode parent, bool keep_scene_transform);
+
+    /*#
+     * Query if the node is a bone
```

Comment:
Why do we still need this in the API? Isn't this something specific to spine?

Reply:
It is a very good question, and it turns out we don't have the need for this function anymore (I removed all the code).

The reason I kept it was most likely since the gameobjects treat them the bones specially (I think that code should be possible to be removed as well)